server/tests/test_migrations.py

```diff
@@ -49,9 +49,399 @@ def pg_psql(pg_cluster: Cluster) -> str:
     return pg_cluster._find_pg_binary("psql")  # type: ignore[attr-defined]
 
 
+def collect_data_patches() -> tuple[dict[int, str], dict[int, str]]:
+    PATCH_FILE_PATTERN = r"^(?P<id>\d{4})_(?P<type>before|after).sql$"
+    from tests import migrations as migrations_test_module
+
+    before = {}
+    after = {}
+    for file in importlib.resources.files(migrations_test_module).iterdir():
+        file_name = file.name
+        match = re.search(PATCH_FILE_PATTERN, file_name)
+        assert match or not file_name.endswith(
+            ".sql"
+        ), f"unknown file `{file_name}`, expects only `xxxx_[before|after].sql`"
+        if match:
+            index = int(match.group("id"))
+            if match.group("type") == "before":
+                patches = before
+            else:  # after
+                patches = after
+
+            assert index not in patches  # Sanity check
+            sql = importlib.resources.files(migrations_test_module).joinpath(file_name).read_text()
+            patches[index] = sql
+
+    return before, after
+
+
+from tests.common import Backend, CoolorgRpcClients
+
+
 @pytest.mark.postgresql
 @pytest.mark.asyncio
-async def test_migrations(pg_cluster_url, pg_psql, pg_dump):
+async def test_migrations(
+    pg_cluster_url: str,
+    pg_psql: str,
+    pg_dump: str,
+    coolorg: CoolorgRpcClients,
+    backend: Backend,
+    backend_config,
+):
+    # await backend.block.create(DateTime.now(), organization_id=coolorg.organization_id, author=coolorg.alice.device_id , realm_id=coolorg.wksp1_id, block_id=BlockID.new(), key_index=1, block=b"block 1")
+    # await backend.block.create(DateTime.now(), organization_id=coolorg.organization_id, author=coolorg.alice.device_id , realm_id=coolorg.wksp1_id, block_id=BlockID.new(), key_index=1, block=b"block 2")
+
+    # zack_u = UserCertificate(
+    #     author=coolorg.alice.device_id,
+    #     timestamp=DateTime.now(),
+    #     user_id=UserID.new(),
+    #     human_handle=HumanHandle("zack@invalid.com", "Zack"),
+    #     public_key=PrivateKey.generate().public_key,
+    #     algorithm=PrivateKeyAlgorithm.X25519_XSALSA20_POLY1305,
+    #     profile=UserProfile.STANDARD,
+    # )
```

Comment:
Debug artefacts?

Reply:
yup, removed (well moved to another file given I think it's still useful to have this at hand whenever we want to generate data or do copy/pasting)